core/overdrive.py

```diff
@@ -231,31 +231,48 @@ def __init__(self, _db: Session, collection: Collection):
         self._external_integration = collection.external_integration
         self._collection_id = collection.id
 
+        # Initialize configuration information.
+        self._configuration_storage = ConfigurationStorage(self)
+        self._configuration_factory = ConfigurationFactory()
+        self._configuration = OverdriveConfiguration(
+            configuration_storage=self._configuration_storage, db=_db
+        )
+
         if collection.parent:
             # This is an Overdrive Advantage account.
             self.parent_library_id = collection.parent.external_account_id
 
             # We're going to inherit all of the Overdrive credentials
             # from the parent (the main Overdrive account), except for the
             # library ID, which we already set.
-            collection = collection.parent
+            parent_integration = collection.parent.external_integration
+
+            self._configuration.client_key = parent_integration.setting(
+                "overdrive_client_key"
+            )
+            self._configuration.client_password = parent_integration.setting(
+                "overdrive_client_secret"
+            )
+            self._configuration.website_id = parent_integration.setting(
+                "overdrive_website_id"
+            )
```

Comment:
The `_server_nickname`, which is one of the settings from the parent collection is not captured in the configuration here.

With the new configuration approach, is there a way to get *all* of the parent collection's settings without enumerating them individually?

Reply:
We seem to have switched from using the class variables (e.g., "SERVER_NICKNAME", "WEBSITE_ID")  for the settings keys here. Because we also reference these keys multiple times in the tests, I think it is worthwhile to update them to their new values (or add them, in the case of the CLIENT_KEY and CLIENT_SECRET) and use them here and in the tests.

Reply:
> With the new configuration approach, is there a way to get all of the parent collection's settings without enumerating them individually?

I think we can instantiate an `OverdriveConfiguration` value from the parent collection and copy settings that way. I'll try it.

Reply:
> We seem to have switched from using the class variables (e.g., "SERVER_NICKNAME", "WEBSITE_ID") for the settings keys here. Because we also reference these keys multiple times in the tests, I think it is worthwhile to update them to their new values (or add them, in the case of the CLIENT_KEY and CLIENT_SECRET) and use them here and in the tests.

Yeah, I agree. The old configuration system uses these generic names in a vain attempt to provide a sort of common API over the different external integration types. The end result of which is that you end up with a dozen rows in the configuration settings table called `username`, and you have to work backwards to determine "Oh, _this_ `username` is an Overdrive client key" and so on. I'd much rather remove the generic names entirely and introduce a set of new `OVERDRIVE_*` class variables.

Reply:
We could tell which `username` is a an OD client key because the `externalintegrations.protocol` is "Overdrive" and the `configurationsettings.key` was "username". These changes certainly give you a hint if you're looking at the database directly. But one value of the class constants is to avoid typos and the like and to allow us to change a value in a single place for, say, a database migration.

Reply:
> We could tell which username is a an OD client key because the externalintegrations.protocol is "Overdrive" and the configurationsettings.key was "username".

It's that extra required step of backwards reasoning that I dislike, though. I'd rather be explicit than general, for the sake of my own and future developer's understanding. Now, if a developer wants to find an Overdrive client key, they look directly at the entries in the table that say `overdrive_client_key`.